File: project_3/imp_1_bruteforcing/main.go
package main

import (
	"fmt"
	"math"

	"github.com/schollz/progressbar/v3"
)

func main() {

	const number float64 = 600851475143
	halfway := math.Floor(number / 2)
	bar := progressbar.Default(int64(halfway))

	for i := halfway; i >= 2; i-- {
		bar.Add(1)
		if int(number)%int(i) == 0 {
			if isPrime(int(i)) {
				fmt.Println(i)
				break
			}

		}

	}
}

func isPrime(n int) bool {

	if n <= 1 {
		return false
	}

	for i := 2; i < n; i++ {
		if n%i == 0 {
			return false
		}

	}
	return true
}

File: project_3/imp_1_bruteforcing/main.js
function isPrime(x) {
    if (x <= 1) {
        return false;
    }
    for (var i = 2; i < x; i++) {
        if (x % i === 0) {
            return false;
        }
    }
    return true;
}
var number = 600851475143;
var halfWay = Math.floor(number / 2);
for (var i = halfWay; function (i) { return 2; }; --i) {
    if (number % i === 0) {
        if (isPrime(i)) {
            console.log(i);
            break;
        }
    }
}
